pkg/api/v1/clients/kube/resource_client.go

```diff
@@ -377,6 +392,20 @@ func (rc *ResourceClient) exist(ctx context.Context, namespace, name string) boo
 
 }
 
+func (rc *ResourceClient) convertCrdToInvalidResource(resourceCrd *v1.Resource) (resources.Resource, error) {
+	resourceCrd.Status.State = core.Status_Invalid
+	_, err := rc.crdClientset.ResourcesV1().Resources(resourceCrd.Namespace).Update(resourceCrd)
```

Comment:
reporting the status to storage should probably happen in another function

also, that function should be gated behind a parameter for the constructor like `EnableErrorReporting`